src/parquet/column/writer.cc

```diff
@@ -134,52 +141,72 @@ void ColumnWriter::WriteDataPage(const DataPage& page) {
 int64_t ColumnWriter::Close() {
   if (!closed_) {
     closed_ = true;
-    if (has_dictionary_) { WriteDictionaryPage(); }
+    if (has_dictionary_ && !fallback_) { WriteDictionaryPage(); }
```

Comment:
Is this correct? You write dictionary index pages until the dictionary exceeds a particular size. You still need the dictionary